integration-tests-ex/docs/guide.md

```diff
@@ -239,3 +239,29 @@ test. Your test will run in parallel with all other IT categories, which is why
 we offered the advice above: the test has to have a good reason to fire up yet
 another build task.
 
+### Choosing the Middle Manager or Indexer
+
+Tests should run on the Middle Manager by default. Tests can optionally runn on the
+Indexer. To run on Indexer:
+
+* In the environment, `export DRUID_INTEGRATION_TEST_INDEXER=indexer`. (Use `middleManager`
+  otherwise. If the variable is not set, `middleManager` is the default.)
+* The `cluster/<category>/docker-compose.yaml` file should be for the Middle manager. Create
+  a separate file called `cluster/<category>/docker-compose-indexer.yaml` to define the
+  Indexer-based cluster.
+* The test `src/test/resources/cluster/<category>/docker.yaml` file should contain a conditional
```

Comment:
I have a test case that should run only on MM. 
As this test case is part of the MSQ category, how can I control to skip running this test . 
The test is namely `ITMultiStageQueryWorkerFaultTolerance`. I had to create a new category for it. 

The ideal way would be to control "not running a particular test" at each test case level. 
By default, the test case runs on both MM's and indexers unless specified otherwise. 
Thoughts?

Reply:
The current structure is as follows:

* The `integration-tests-ex/cases` module contains all the IT tests.
* Tests are grouped into a "category". The category defines the set of JUnit test to run _and_ the cluster to use to run the tests.
* A test category can "reuse" a cluster definition from some other category, if no changes are needed.
* With this PR, each test category can have an Indexer and Middle Manager configuration (as distinct files.)

If I understand correctly, you have an MSQ test that can't run on Indexer, only MM. In this case, the fourth bullet (run on both) isn't needed.

The default solution is that you'd define your `docker-compose.yaml` file to include MM. You _would not_ provide a `docker-compose-indexer.yaml` file. When the `it.sh` script tries to run your test, with `DRUID_INTEGRATION_TEST_INDEXER=indexer`, the test will fail because of the missing `docker-compose-indexer.yaml` file.

This should be OK. In Travis, we have a separate entry for each IT: indexer or not. We'd not include a line for your MSQ test with "indexer".

If we simply want to succeed, but do nothing, then add the following lines to `cluster.sh` at line 156:

```bash
			if [ ! -f "$CLUSTER_DIR/$compose_file" ]; then
                           if [ $CATEGORY == "SomeMSQTest" ]; then
                             exit 0 # Silently succeed for SomeMSQTest with indexer
                           fi
```

Reply:
@paul-rogers As I understand the above question - 
Say we have 10 tests in the `MSQ` category - 9 of them run for both MM and indexer and one test only runs on MM.
For that one test - do we need to create a new category (which only has `docker-compose.yaml` file) or can we have the option to skip it when the tests MSQ category tests run with indexer?

Both `docker-compose.yaml` and `docker-compose-indexer.yaml` will be present as the other tests in the category need it.

Reply:
If you have one category, that includes 10 `@Test` cases, and one of those test cases can't run in Indexer, then the easiest solution is for us to provide an `isIndexer()` method. The test case can be written as:

```java
\@Test
public void myMMOnlyTest()
{
  if (TestUtils.isIndexer()) {
    // This test can't run on indexer
    return
  }
  // MM-only test case here
}
```

We could make this fancy, by defining an annotation, and using our `DruidTestRunner` to do filtering.

The implementation of `isIndexer()` is simply `"indexer".equals(System.getenv("DRUID_INTEGRATION_TEST_INDEXER"))`.